### DISCUSSION TITLE
Job release with delay not working as expected. L9 & Beanstalkd

### DISCUSSION THREAD

**Discussion by rbruhn** [laravel/framework#43791] [Q&A]
Laravel 9 & Beanstalkd

When trying to release a job back to the queue with a delay, it is ignored. Below is the logging, supervisor command, and test code I'm using. Previously, this worked in Laravel. Watching the Beanstalk console, I see the job run once and it's simply done. Never runs again.

Did something change lately? Or am I missing something?

Log
```
[2022-08-20 20:15:15] dev.INFO: Attempt 1  
[2022-08-20 20:15:15] dev.INFO: releasing attempt 1  
```

Supervisor
```
command=php /home/vagrant/sites/my_site/artisan queue:work beanstalkd --force --queue=export --tries=3
```

Code
```
class TestJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    public int $tries = 3;

    /**
     * Create a new job instance.
     *
     * @return void
     */
    public function __construct()
    {
        $this->onQueue(config('config.export_tube'));
    }

    /**
     * Execute the job.
     *
     * @return void
     */
    public function handle()
    {
        \Log::info('Attempt ' . $this->attempts());
        if ($this->attempts() < 3) {
            \Log::info('releasing attempt ' . $this->attempts());
            $this->release(30);

            return;
        }

        \Log::info('Done releasing');

    }
}

```

Edit: The above works correctly. Apparently it's some issue when using my testing subdomain even thought it's the same code.

**Comment by romuloMendes** [comment]
Maybe this works.

`
namespace App\Jobs;

use Illuminate\Bus\Queueable;
use Illuminate\Contracts\Queue\ShouldQueue;
use Illuminate\Foundation\Bus\Dispatchable;
use Illuminate\Queue\InteractsWithQueue;
use Illuminate\Queue\SerializesModels;
use Illuminate\Support\Facades\Log;

class ParentJob implements ShouldQueue
{
    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;

    public int $tries = 3;

    public function handle()
    {
        // Log::info('Attempt ' . $this->attempts());
        if ($this->attempts() < 3) {
            Log::info('releasing attempt ' . $this->attempts());
            $this->release(5);

            return;
        }

        Log::info('Done releasing');
    }

    public function retryUntil()
    {
        // will keep retrying, by backoff logic below
        // until 12 hours from first run.
        // After that, if it fails it will go
        // to the failed_jobs table
        Log::info('retryUntil ' . now()->addSeconds(3)->format('Y-m-d H:i:s'));

        return now()->addSeconds(3);
    }

    /**
     * Calculate the number of seconds to wait before retrying the job.
     *
     * @return array
     */
    public function backoff()
    {
        // first 5 retries, after first failure
        // will be 5 minutes (300 seconds) apart,
        // further attempts will be
        // 3 hours (10,800 seconds) after
        // previous attempt
        Log::info('backoff ' . $this->attempts());
        return [5, 5, 5, 5, 5, 10];
    }
}
`
